[RDY] Fix tests for plain lua

tarruda opened the pull request:
Make it possible to run functional tests with plain lua, which is useful for inspecting the lua stack trace with gdb when tests hang

Also include a fix from #1820

tarruda committed:
test: Fix screen.lua eol_clear method.

tarruda committed:
test: Make it possible to run functional tests with plain lua

Replace the hexadecimal escape sequences, which are only supported by luajit

fwalch commented:
There's a test error:
```
Screen bell/visual bell is forwarded to the UI
./test/functional/ui/screen.lua:182: Visual bell was not sent
```

tarruda committed:
test: Set some options to reduce nondeterminism in functional tests

- shortmess+=I: Remove intro screen
- background=light: Disregard COLORFGBG environment variable
